xilem_web/src/concurrent/memoized_await.rs

```diff
@@ -84,20 +169,78 @@ where
     State: 'static,
     Action: 'static,
     Data: PartialEq + 'static,
-    FOut: std::fmt::Debug + 'static,
+    FOut: fmt::Debug + 'static,
     F: Future<Output = FOut> + 'static,
     InitFuture: Fn(&Data) -> F + 'static,
     OA: OptionalAction<Action> + 'static,
     Callback: Fn(&mut State, FOut) -> OA + 'static,
 {
-    MemoizedAwait {
+    MemoizedAwait(MemoizedInner {
         init_future,
         data,
         callback,
         debounce_ms: 0,
         reset_debounce_on_update: true,
         phantom: PhantomData,
-    }
+    })
+}
+
+/// Await a stream returned by `init_stream` invoked with the argument `data`, `callback` is called with the items of the stream.
+/// `init_stream` will be invoked again, when `data` changes.
+///
+/// The update behavior can be controlled, by [`debounce_ms`](`MemoizedStream::debounce_ms`) and [`reset_debounce_on_update`](`MemoizedStream::reset_debounce_on_update`)
+///
+/// # Examples
+///
+/// ```
+/// use gloo_timers::future::TimeoutFuture;
+/// use async_stream::stream;
+/// use xilem_web::{core::fork, concurrent::memoized_stream, elements::html, interfaces::Element};
+///
+/// fn app_logic(state: &mut Vec<usize>) -> impl Element<Vec<usize>> {
+///     fork(
+///         html::div(format!("{state:?}")),
+///         memoized_stream(
+///             10,
+///             |n| {
+///                 let range = 0..*n;
+///                 stream! {
+///                     for i in range {
+///                         TimeoutFuture::new(500).await;
+///                          yield i;
+///                      }
+///                 }
+///             },
+///             |state: &mut Vec<usize>, item: usize| {
+///                 state.push(item);
+///             },
+///         ),
+///     )
+/// }
+/// ```
```

Comment:
:+1:  on doc-comment.

I wonder whether it makes sense to have something like I've done [here](https://github.com/Philipp-M/trui/blob/dd24a2921eb0587dfffbbafea4a3e62b92f39c43/examples/async_event_handler_stream.rs#L53-L60). To know in what the state of the stream is.

Reply:
I use `StreamMessage::Started` here instead of `StreamMessage::Started(T)` 
because there can be a big difference between the start of the stream and the reception of the first element.
Does that make sense?

Reply:
I've just realized that it's easier and cleaner if you leave the responsibility to the user. For example you might want to add some payload to the `Started` variant. And even existing streams could be wrapped.